### DISCUSSION TITLE
do you guys model your individual IP's with a /32 CIDR? or inherit the CIDR mask of its parent prefix?

### DISCUSSION THREAD

**Discussion by ghost** [netbox-community/netbox#18053] [Q&A]
We have been adding our IP's for individual hosts with a x.x.x.x/32 CIDR.
However, I noticed that when I click on the ## Available IP's link inside a prefix to have NetBox automatically begin creating a new IP Address for me, I notice that it uses whatever CIDR mask the larger subnet uses.

So if my prefix is a /26, when I click the link netbox will use the first available IP with a /26 prefix.

Is this how we're supposed to be modeling individual IP's?

**Comment by candlerb** [comment]
> So if my prefix is a /26, when I click the link netbox will use the first available IP with a /26 prefix.
> 
> Is this how we're supposed to be modeling individual IP's?

Yes. Every time you record an individual IP address in Netbox, you are required to record with it the prefix length of the prefix it belongs to. IMO it's a lot of data duplication, and a lot of opportunity for errors and inconsistencies.

There have been multiple discussions of this in the past - e.g. #800, #8731 - and it comes up from time to time, but that battle has been long lost.

Official answer: https://github.com/netbox-community/netbox/wiki/Frequently-Asked-Questions#why-does-netbox-require-every-ip-to-have-a-mask

**Reply by ghost** [reply]
the question isn't about whether or not we have to add a CIDR mask. The question is which CIDR mask should we use.

> Every time you record an individual IP address in Netbox, you are required to record with it the **prefix length of the prefix it belongs to**.

Say's who?
NetBox allows us to enter the IP with a /32 or a /24 or really any mask .. as long as we add a mask.
If we have a /24 prefix, and we add an individual IP address and use a /32 for that IP rather than the /24 of its parent prefix, what are the ramifications of that?

to be clear, I'm fine following what you say above. We can easily go bulk update all our IP's to reflect the CIDR mask of its parent prefix, I'm just curious why it is the recommendation to use the parent prefixes CIDR rather than a /32 in the context of a single individual IP address.

**Reply by etfz** [reply]
I suppose it's meant to document which subnet mask you're using on the host for that address. If we were supposed to to put /32 for any host there would be no point in requiring a prefix length for individual IP addresses.
